cli/azd/cmd/middleware/ux.go

```diff
@@ -0,0 +1,36 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/pkg/operations"
+)
+
+// UxMiddleware is a middleware that handles the UX of the CLI
+type UxMiddleware struct {
+	options          *Options
+	operationPrinter operations.Printer
+}
+
+// Creates a new instance of the UX middleware
+func NewUxMiddleware(options *Options, operationPrinter operations.Printer) Middleware {
+	return &UxMiddleware{
+		options:          options,
+		operationPrinter: operationPrinter,
+	}
+}
+
+func (m *UxMiddleware) Run(ctx context.Context, next NextFn) (*actions.ActionResult, error) {
+	// Don't run for sub actions
+	if m.options.IsChildAction() {
+		return next(ctx)
+	}
+
+	if err := m.operationPrinter.Start(ctx); err != nil {
```

Comment:
Using the middleware here so that our operations printer can receive and process messages at the hooks level as well as commands.